File: 78-subsets/78-subsets.cpp
class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        
        int n = nums.size();
        int powsize = 1<<n;
        vector<vector<int>> res(powsize);
        
        for(int i=0;i<powsize;i++)
        {
            for(int j=0;j<n;j++)
            {
                if(i & (1<<j))
                {
                    res[i].push_back(nums[j]);
                }
            }
        }
        return res;
    
    }
};

File: 78-subsets/78-subsets.java
class Solution {
    public List<List<Integer>> subsets(int[] nums) {
        List<List<Integer>> ans = new ArrayList<>();
        List<Integer> cs = new ArrayList<>();
        sset(nums,0,cs,ans);
        return ans;
    }
    private void sset(int[] nums, int ci,List<Integer> currentsubset, List<List<Integer>> ans)
    {
        if(ci >= nums.length)
        {
            ans.add(new ArrayList<>(currentsubset));
            return ;
        }
        
        currentsubset.add(nums[ci]);
        sset(nums,ci+1,currentsubset,ans);
        
        currentsubset.remove(currentsubset.size()-1);
        sset(nums,ci+1,currentsubset,ans);
        return ;
    }
}